### PR TITLE
examples/suit_update: add ble documentation in README

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

This PR extends the doc to show how to perform `suit_updates` over ble. Two cases are added:

- RIOT ble border router + ble suit-device
- Linux ble host + ble-device

We are using `autoconn` so the setup is minimal and no changes to the code are necessary.

NOTE: I had some issues with my local driver for BLE, I ended up using a 4.0 bluetooth usb dongle and it worked straight away.

- 0e3097a is the main commit to review
- 7a911dc might be included in #12361
- eb6522a will be PR'd separately.

### Testing procedure

Testing here should mean following the doc and being able to perform suit-updates in both scenarios. 

I would recommend rebasing on #11553 so the compilation of the pkg's isn't triggered for every `flash`, `publish`.

### Issues/PRs references

Depends on ~#11818~ and #12361

### CODE REVIEW

**Comment by fjmolinas** [comment]
@emmanuelsearch might want to give it a test.

**Comment by emmanuelsearch** [comment]
@fjmolinas needs rebase now that #12361 is merged?

**Comment by aabadie** [comment]
please rebase!

**Comment by emmanuelsearch** [comment]
@fjmolinas @aabadie seems there are no blockers to merging this PR, right?

**Comment by fjmolinas** [comment]
> @fjmolinas @aabadie seems there are no blockers to merging this PR, right?

No, I haven't gone back to this because I do not have the HW to test at the moment.

**Comment by fjmolinas** [comment]
> No, I haven't gone back to this because I do not have the HW to test at the moment.

To test the linux part.

File: examples/suit_update/README.md [hunk 1]
```diff
@@ -185,10 +202,130 @@ In this case the RIOT node can be reached from the host using its global address
...
+            RX errors 0  dropped 0  overruns 0  frame 0
+            TX packets 354  bytes 30618 (30.6 KB)
+            TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
+    ...
+
+- You should now be able to ping the device
+
+    $ ping6 fe80::e4dd:e0ff:fe8f:7365%bt0
+
+If the optional configuration to set-up a routable prefix is not performed then
```
Comment (aabadie) [inline examples/suit_update/README.md]: This paragraph needs the be rewritten.

File: examples/suit_update/README.md [hunk 2]
```diff
@@ -185,10 +202,130 @@ In this case the RIOT node can be reached from the host using its global address
...
+            RX packets 3  bytes 120 (120.0 B)
+            RX errors 0  dropped 0  overruns 0  frame 0
+            TX packets 34  bytes 3585 (3.5 KB)
+            TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
+
+In this case the address to use for `SUIT_COAP_SERVER` can be either the EUI64
+generated global address `[2001:db8::19:86ff:fe00:16ca]` or the random global address
+`[2001:db8::b004:c58:891f:aa09]`.
+
+I for some reason this didn't work the you can manually set up an address for
```
Comment (aabadie) [inline examples/suit_update/README.md]:
```suggestion
If for some reason this didn't work the you can manually set up an address for
```

File: examples/suit_update/README.md [hunk 3]
```diff
@@ -185,10 +202,130 @@ In this case the RIOT node can be reached from the host using its global address
...
+
+In either case the address used for `SUIT_CLIENT` should be the suffix of the link
+local address for that device (`e4dd:e0ff:fe8f:7365` in our examples) and the
+distributed prefix, i.e.: `SUIT_CLIENT=[2001:db8::e4dd:e0ff:fe8f:7365]`
+
 ### Start aiocoap-fileserver
 [Start-aiocoap-fileserver]: #start-aiocoap-fileserver
 
-`aiocoap-fileserver` is used for hosting firmwares available for updates.
+`aiocoap-fileserver` is used for hosting firmware's available for updates.
```
Comment (aabadie) [inline examples/suit_update/README.md]: A bit unrelated.
```suggestion
`aiocoap-fileserver` is used for hosting the firmwares available for updates.
```

File: examples/suit_update/README.md [hunk 4]
```diff
@@ -11,6 +11,7 @@ the manifest format specified in
 Table of contents:
 
 - [Prerequisites][prerequisites]
+  - [Ble Prerequisites][prerequisites-ble]
```
Comment (aabadie) [inline examples/suit_update/README.md]:
```suggestion
  - [Ble][prerequisites-ble]
```

File: examples/suit_update/README.md [hunk 5]
```diff
@@ -63,6 +65,12 @@ Table of contents:
   using `make term`, but that requires an already set up tap interface.
   See [update] for more information.
 
+
+### Ble Prerequisites
```
Comment (aabadie) [inline examples/suit_update/README.md]:
```suggestion
### Ble
```

File: examples/suit_update/README.md [hunk 6]
```diff
@@ -63,6 +65,12 @@ Table of contents:
...
+[prerequisites-ble]: #BlePrerequisites
```
Comment (aabadie) [inline examples/suit_update/README.md]:
```suggestion
[prerequisites-ble]: #Ble
```

**Comment by aabadie** [comment]
ping @fjmolinas :)

**Comment by fjmolinas** [comment]
Re tested node to node but I can't seem to make the linux part work.
<details>

```
> 2020-01-14 16:37:50,930 #  help
2020-01-14 16:37:50,933 # Command              Description
2020-01-14 16:37:50,937 # ---------------------------------------
2020-01-14 16:37:50,940 # reboot               Reboot the node
2020-01-14 16:37:50,943 # ping6                Ping via ICMPv6
2020-01-14 16:37:50,947 # random_init          initializes the PRNG
2020-01-14 16:37:50,952 # random_get           returns 32 bit of pseudo randomness
2020-01-14 16:37:50,957 # nib                  Configure neighbor information base
2020-01-14 16:37:50,961 # ifconfig             Configure network interfaces
2020-01-14 16:37:50,966 # 6ctx                 6LoWPAN context configuration tool
2020-01-14 16:37:50,971 # ble                  Manage BLE connections for NimBLE
2020-01-14 16:37:50,976 # suit                 Trigger a SUIT firmware update
> ifconfig
2020-01-14 16:37:53,148 #  ifconfig
2020-01-14 16:37:53,151 # Iface  8  HWaddr: D8:D1:62:91:91:33 
2020-01-14 16:37:53,156 #           L2-PDU:1280 MTU:1280  HL:64  RTR  
2020-01-14 16:37:53,157 #           6LO  IPHC  
2020-01-14 16:37:53,161 #           Source address length: 6
2020-01-14 16:37:53,163 #           Link type: wireless
2020-01-14 16:37:53,169 #           inet6 addr: fe80::d8d1:62ff:fe91:9133  scope: link  VAL
2020-01-14 16:37:53,172 #           inet6 group: ff02::2
2020-01-14 16:37:53,175 #           inet6 group: ff02::1
2020-01-14 16:37:53,178 #           inet6 group: ff02::1:ff91:9133
2020-01-14 16:37:53,179 #           
> ifconfig
2020-01-14 16:39:38,308 #  ifconfig
2020-01-14 16:39:38,312 # Iface  8  HWaddr: D8:D1:62:91:91:33 
2020-01-14 16:39:38,316 #           L2-PDU:1280 MTU:1280  HL:64  RTR  
2020-01-14 16:39:38,319 #           RTR_ADV  6LO  IPHC  
2020-01-14 16:39:38,322 #           Source address length: 6
2020-01-14 16:39:38,325 #           Link type: wireless
2020-01-14 16:39:38,331 #           inet6 addr: fe80::d8d1:62ff:fe91:9133  scope: link  VAL
2020-01-14 16:39:38,337 #           inet6 addr: 2001:db8::d8d1:62ff:fe91:9133  scope: global  VAL
2020-01-14 16:39:38,340 #           inet6 group: ff02::2
2020-01-14 16:39:38,343 #           inet6 group: ff02::1
2020-01-14 16:39:38,346 #           inet6 group: ff02::1:ff91:9133
2020-01-14 16:39:38,347 #           
> 2020-01-14 16:42:49,399 #  suit: received URL: "coap://localhost/fw/nrf52dk/suit_update-riot.suitv4_signed.latest.bin"
2020-01-14 16:42:49,402 # suit_coap: trigger received
2020-01-14 16:42:49,410 # suit_coap: downloading "coap://localhost/fw/nrf52dk/suit_update-riot.suitv4_signed.latest.bin"
2020-01-14 16:42:49,411 # suit: invalid URL
2020-01-14 16:42:49,414 # suit_coap: error getting manifest
2020-01-14 16:46:35,350 # suit: received URL: "coap://localhost/fw/nrf52dk/suit_update-riot.suitv4_signed.latest.bin"
2020-01-14 16:46:35,354 # suit_coap: trigger received
2020-01-14 16:46:35,362 # suit_coap: downloading "coap://localhost/fw/nrf52dk/suit_update-riot.suitv4_signed.latest.bin"
2020-01-14 16:46:35,364 # suit: invalid URL
2020-01-14 16:46:35,366 # suit_coap: error getting manifest
2020-01-14 16:47:17,272 # suit: received URL: "coap://[2001:db8::1]/fw/nrf52dk/suit_update-riot.suitv4_signed.latest.bin"
2020-01-14 16:47:17,274 # suit_coap: trigger received
2020-01-14 16:47:17,283 # suit_coap: downloading "coap://[2001:db8::1]/fw/nrf52dk/suit_update-riot.suitv4_signed.latest.bin"
2020-01-14 16:47:18,691 # suit_coap: got manifest with size 525
2020-01-14 16:47:18,695 # suit: verifying manifest signature...
2020-01-14 16:47:20,659 # suit: validated manifest version
2020-01-14 16:47:20,664 # )riotboot_hdr_validate: riotboot_hdr magic number invalid
2020-01-14 16:47:20,667 # suit: validated sequence number
2020-01-14 16:47:20,668 # )validating vendor ID
2020-01-14 16:47:20,678 # Comparing 547d0d74-6d3a-5a92-9662-4881afd9407b to 547d0d74-6d3a-5a92-9662-4881afd9407b from manifest
2020-01-14 16:47:20,680 # validating vendor ID: OK
2020-01-14 16:47:20,682 # validating class id
2020-01-14 16:47:20,690 # Comparing c7405d52-8e95-5611-9494-9b8be4cf74a3 to c7405d52-8e95-5611-9494-9b8be4cf74a3 from manifest
2020-01-14 16:47:20,692 # validating class id: OK
2020-01-14 16:47:20,698 # Comparing manifest offset 8192 with other slot offset 266240
2020-01-14 16:47:20,704 # Comparing manifest offset 266240 with other slot offset 266240
2020-01-14 16:47:20,709 # Comparing manifest offset 8192 with other slot offset 266240
2020-01-14 16:47:20,715 # Comparing manifest offset 266240 with other slot offset 266240
2020-01-14 16:47:20,720 # riotboot_flashwrite: initializing update to target slot 1
2020-01-14 16:47:20,864 # Fetching firmware |█████████████████████████| 100%
2020-01-14 16:53:43,115 # riotboot: verifying digest at 0x200038db (img at: 0x41000 size: 157940)
2020-01-14 16:53:43,340 # suit_v4_parse() success
2020-01-14 16:53:43,342 # SUIT policy check OK.
2020-01-14 16:53:43,345 # suit_coap: finalizing image flash
2020-01-14 16:53:43,424 # riotboot_flashwrite: riotboot flashing completed successfully
2020-01-14 16:53:43,427 # Image magic_number: 0x544f4952
2020-01-14 16:53:43,429 # Image Version: 0x5e1de24a
2020-01-14 16:53:43,432 # Image start address: 0x00041400
2020-01-14 16:53:43,434 # Header chksum: 0x2902f20d
2020-01-14 16:53:43,435 # 
2020-01-14 16:53:44,700 # main(): This is RIOT! (Version: 2020.01-devel-1727-gad661-pr_suit_ble)
2020-01-14 16:53:44,704 # RIOT SUIT update example application
2020-01-14 16:53:44,705 # running from slot 1
2020-01-14 16:53:44,708 # slot start addr = 0x41000
2020-01-14 16:53:44,710 # Image magic_number: 0x544f4952
2020-01-14 16:53:44,713 # Image Version: 0x5e1de24a
2020-01-14 16:53:44,716 # Image start address: 0x00041400
2020-01-14 16:53:44,717 # Header chksum: 0x2902f20d
```
</details>

**Comment by fjmolinas** [comment]
> Re tested node to node but I can't seem to make the linux part work.

EDIT: the radvd part only.

**Comment by fjmolinas** [comment]
@aabadie I had to add the route for `bt0` for it to work, you have some experience with `radvd`, any idea why?

`sudo route -A inet6 add 2001:db8::/64 dev bt0`

**Comment by fjmolinas** [comment]
@aabadie I think I addressed your comments and worked on linux as well.

File: examples/suit_update/README.md [hunk 7]
```diff
@@ -185,10 +202,140 @@ In this case the RIOT node can be reached from the host using its global address
...
+
+Route traffic going towards your subnet through bt0:
+
+    $ sudo route -A inet6 add 2001:db8::/64 dev bt0
+
+In either case the address used for `SUIT_CLIENT` should be the suffix of the link
+local address for that device (`e4dd:e0ff:fe8f:7365` in our examples) and the
+distributed prefix, i.e.: `SUIT_CLIENT=[2001:db8::e4dd:e0ff:fe8f:7365]`
+
+If this optional step is skipped then in then `SUIT_COAP_SERVER` will be
```
Comment (aabadie) [inline examples/suit_update/README.md]:
```suggestion
If this optional step is skipped then `SUIT_COAP_SERVER` will be
```

File: examples/suit_update/README.md [hunk 8]
```diff
@@ -185,10 +202,140 @@ In this case the RIOT node can be reached from the host using its global address
...
+the link local address of the `bt0` interface and `SUIT_CLIENT` will be
+link local address of the device, with the interface specified. e.g:
```
Comment (aabadie) [inline examples/suit_update/README.md]:
```suggestion
the link local address of the device, with the interface specified. e.g:
```

File: examples/suit_update/README.md [hunk 9]
```diff
@@ -185,10 +202,140 @@ In this case the RIOT node can be reached from the host using its global address
...
+            RX packets 3  bytes 120 (120.0 B)
+            RX errors 0  dropped 0  overruns 0  frame 0
+            TX packets 34  bytes 3585 (3.5 KB)
+            TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
+
+In this case the address to use for `SUIT_COAP_SERVER` can be either the EUI64
+generated global address `[2001:db8::19:86ff:fe00:16ca]` or the random global address
+`[2001:db8::b004:c58:891f:aa09]`.
+
+If for some reason this didn't work the you can manually set up an address for
```
Comment (aabadie) [inline examples/suit_update/README.md]:
```suggestion
If for some reason this didn't work, you can manually set up an address for
```

File: examples/suit_update/README.md [hunk 10]
```diff
@@ -185,10 +202,140 @@ In this case the RIOT node can be reached from the host using its global address
...
+      ...
+      ...
+          [bluetooth]# devices
+          Device F0:36:27:6B:F1:8F Decathlon Dual HR
+          Device 69:B3:82:0B:73:C9 69-B3-82-0B-73-C9
+          Device 43:1A:39:CD:39:B9 43-1A-39-CD-39-B9
+          Device E4:DD:E0:8F:73:65 RIOT-autoconn
+      ...
+
+- Once you have the address, you simply connect Linux to RIOT using the following
```
Comment (aabadie) [inline examples/suit_update/README.md]:
```suggestion
- Once you have the address, simply connect Linux to RIOT using the following
```

**Pull request merged**